README.md

```diff
@@ -27,10 +31,18 @@ class Vehicle(vqpy.VObjBase):
         return self.infer('license_plate', {'license_plate': 'openalpr'})
 ```
 
+Here you can see a builtin interface `infer` provided by the library. There are two basic interfaces `infer` and `getv`, both of which are used to infer an attribute, typically an user-defined or pre-defined property in most cases. There are two differences in the functions: `getv` can retrieve values in historic frames if the property is marked by `@vqpy.stateful` decorator; `getv` consider user-defined names first and `infer` directly look up the library.
+
+We also have some other decorators for defining properties, including `@vqpy.postproc` and `@vqpy.cross_object_property`. The former decorator provides basic postprocessing functions. For instance, returning the modal value of the results in past 10 frames. The latter decorator can help user define cross-object properties, so that users can track values like distances between all pairs of people and baggage.
+
+For more details on how to use the decorators to define one's own object, please refer to our API document. You can also refer to our demos to see several example implementations of concerned objects.
+
 #### Define a `Query`
 
 Users can express their queries through SQL-like constraints with `VObjConstraint`, which is a return value of the `setting` method in their `Query` class. In `VObjConstraint`, users can specify query constraints on the interested object with `filter_cons`, and `select_cons` gives the projection of the properties the query shall return.
 
+Moreover, the user can provide some functions when The projected properties will pass a postprocess function provided by the user, which is the identity function by default. 
```

Comment:
Shall we explain that after our example code, since it might be hard for users to realize you are talking about `select_cons`? 
like in `select_cons`, you can add your postprocessing function as the corresponding value of a "property_name" key, and we will output the post-processed value of the "property_name". And you can just set the value to None if you want to directly output the property value.